### PR TITLE
Poll with Current Active Filters

### PR DESCRIPTION
**Pull request by montlebalm**
#### What's this PR do?

This PR fixes the use of `CursorPoller` in `Stream` so that it's updated with the current active filters. Previously, the poller locked in whatever filters were present on page load.

This fixes "Switching the active filter does not prevent old results from loading" on issue #1585.

Change summary:
- StreamManager: Don't add items unless they have an id
- Stream: sync state with url on mount
- CursorPoller: add `setEndpoint` method to update internal url
- Unmount React components after each test
#### Where should the reviewer start?

`src/sentry/static/sentry/app/views/stream.jsx`
`src/sentry/static/sentry/app/utils/cursorPoller.jsx`
#### How should this be manually tested?

Run `make test-js`

You can also start the Sentry server, navigate to the stream page, and select one of the filters (i.e., "Bookmarks", "Assigned") before the poller has a chance to complete its first poll. If the filtered data remains the same after the first poll has completed then the changes have fixed the issue.
#### Any background context you want to provide?

This PR builds upon #1602.
#### What gif best describes this PR or how it makes you feel?

cheers

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/streamManager.jsx [hunk 1]
```diff
@@ -24,6 +24,8 @@ class StreamManager {
     items = [].concat(items);
     if (items.length === 0) return this;
 
+    items = items.filter((item) => item.hasOwnProperty("id"));
```
Comment (dcramer) [inline src/sentry/static/sentry/app/utils/streamManager.jsx]: what was this filtering out?

**Comment by dcramer** [comment]
just that one question otherwise lgtm

File: src/sentry/static/sentry/app/utils/streamManager.jsx (hunk 1 above)
Comment (montlebalm) [inline src/sentry/static/sentry/app/utils/streamManager.jsx]: Without the filter `push` would add `undefined` to the list if the item did not include an `id` property.

Comment (dcramer) [inline src/sentry/static/sentry/app/utils/streamManager.jsx]: Were you actually seeing groups that had no IDs? My main concern is how those would exist

Comment (tracehelms) [inline src/sentry/static/sentry/app/utils/streamManager.jsx]: We didn't see any real instances of it but we just noticed it was possible during testing.

**Pull request merged**